Tribler/Test/Core/CreditMining/test_creditmining_sys.py

```diff
@@ -0,0 +1,553 @@
+
+        self.boosting_manager.add_source(dispersy_cid)
+        chn_obj = self.boosting_manager.get_source_object(dispersy_cid)
+
+        def _load(torrent, callback=None):
+            if not isinstance(torrent, CollectedTorrent):
+                torrent_id = 0
+                if torrent.torrent_id <= 0:
+                    torrent_id = self.session.lm.torrent_db.getTorrentID(torrent.infohash)
+                if torrent_id:
+                    torrent.update_torrent_id(torrent_id)
+
+                torrent = CollectedTorrent(torrent, self.tdef)
+            if callback is not None:
+                callback(torrent)
+            else:
+                return torrent
+
+        chn_obj.torrent_mgr.load_torrent = _load
+
+        def clean_community(_):
+            """
+            cleanly exit the community we are in
+            """
+            if chn_obj.community:
+                chn_obj.community.cancel_all_pending_tasks()
+
+        d = self.check_source(dispersy_cid)
+        d.addCallback(clean_community)
+        return d
+
+    @deferred(timeout=20)
+    def test_chn_max_torrents(self):
+        """
+        Test the restriction of max_torrents in a source.
+        """
+        self.session.get_dispersy = lambda: True
+        self.session.lm.dispersy = Dispersy(ManualEnpoint(0), self.getStateDir())
+        dispersy_cid_hex = "abcd" * 9 + "0012"
+        dispersy_cid = binascii.unhexlify(dispersy_cid_hex)
+
+        # create channel and insert torrent
+        self.create_fake_allchn_comm()
+        self.create_torrents_in_channel(dispersy_cid_hex)
+
+        pioneer_file = os.path.join(TESTS_DATA_DIR, "Pioneer.One.S01E06.720p.x264-VODO.torrent")
+        pioneer_tdef = TorrentDef.load(pioneer_file)
+        pioneer_ihash = binascii.unhexlify("66ED7F30E3B30FA647ABAA19A36E7503AA071535")
+
+        torrent_list = [[self.channel_id, 1, 1, pioneer_ihash, 1460000001, pioneer_file,
+                         pioneer_tdef.get_files_as_unicode_with_length(), pioneer_tdef.get_trackers_as_single_tuple()]]
+        self.insert_torrents_into_channel(torrent_list)
+
+        self.boosting_manager.add_source(dispersy_cid)
+        chn_obj = self.boosting_manager.get_source_object(dispersy_cid)
+        chn_obj.max_torrents = 2
+        chn_obj.torrent_mgr.load_torrent = lambda dummy_1, dummy_2: None
+
+        def _load(torrent, callback=None):
+            if not isinstance(torrent, CollectedTorrent):
+                torrent_id = 0
+                if torrent.torrent_id <= 0:
+                    torrent_id = self.session.lm.torrent_db.getTorrentID(torrent.infohash)
+                if torrent_id:
+                    torrent.update_torrent_id(torrent_id)
+
+                infohash_str = binascii.hexlify(torrent.infohash)
+                torrent = CollectedTorrent(torrent, self.tdef if infohash_str.startswith("fc") else pioneer_tdef)
+            if callback is not None:
+                callback(torrent)
```

Comment:
@ardhipoetra if it's not there already, can you create an issue for this (moving `TorrentManager.loadTorrent` to the core)?